DEVELOPMENT.md

```diff
@@ -0,0 +1,106 @@
+# Development
+
+This document is __a work-in-progress__ which aims to describe various tips for __developing and working with this repository itself__.  It is not intended as a guide on how to use the project within another project, which should be covered elsewhere in the project's documentation.
+
+For now, this will be a collection of tips and tricks for the maintainers of the project.
+
+## Staging a release as a PR
+
+Depending on the size of the release, it may be ideal to have a staging PR which gathers the various features we're intending on releasing into the version.  This can be a great way for the community to understand the scope of a release and a clear way for maintainers to publicize it and gather feedback.
+
+> Note: The instructions here cater to a world where the packages in a repository are published in lock-step, as is the case with all of the `apollo-server-*` packages in the Apollo Server repository.  Release branches are still a good idea, but in this repository it quite possible that there are multiple versions involved (e.g., a version of `@apollo/federation` and a version of `@apollo/gateway`).  In that regard, it's expected that the branch names and the PR title will deviate from the suggestion below.  The branch should still be prefixed with `release-` (for _Branch protection rules_ to match easily), but the rest of the branch name might be different, e.g., `release-federation-X.Y.Z`.
+
+1. Create a branch off `main` named `release-X.Y.Z`, where `X.Y.Z` is the intended release.
+2. Edit the appropriate `CHANGELOG.md` (in the appropriate package directory; e.g., `./federation-js/CHANGELOG.md`, `./gateway-js/CHANGELOG.md`), removing the `vNEXT` section entirely.  This section will remain on `main` and be resolved during the merge.
+3. Add a new section for `### vX.Y.Z` with a bullet indicating that something is coming soon:
+
+   ```
+   ### vX.Y.Z
+
+   - Nothing yet! Stay tuned.
+   ```
+
+4. Commit this change so GitHub will allow a PR to be opened against `main` with a notable change.  A suggested commit message is `Prepare CHANGELOG.md for upcoming vX.Y.Z release.`
+5. Push the branch to GitHub
+6. On GitHub, open a PR from the new release branch which targets `main`.
+   __For the title of the PR__, use "Release X.Y.Z" (See the "Note" above about how this might vary on this repository).  __For the body,__ use the contents of the template in the `.github/APOLLO_RELEASE_TEMPLATE.md` file in this repository.  Adjust the body as you see necessary.
+
+## Publishing a release
+
+### Step 1: Update the appropriate CHANGELOG.md files
+
+There is not a root `CHANGELOG.md` on this monorepo.  Instead, there are two main `CHANGELOG.md` files:
+
+- `./federation-js/CHANGELOG.md` for `@apollo/federation`
+- `./gateway-js/CHANGELOG.md` for `@apollo/gateway`.  __This is also the target CHANGELOG for changes to the `@apollo/query-planner` since that package is nottypically consumed/upgraded directly by our typical user.__
```

Comment:
```suggestion
There is not a root `CHANGELOG.md` on this monorepo.  Instead, there are `CHANGELOG.md` files for specific packages:
 
- `./federation-js/CHANGELOG.md` for `@apollo/federation`
- `./query-planner-js/CHANGELOG.md` for `@apollo/query-planner`

  > Since this is not a direct dependency of our _primary_ consumers, it is best to *also* surface important changes in `@apollo/gateway`'s `CHANGELOG.md`.
- `./gateway-js/CHANGELOG.md` for `@apollo/gateway`.
```